### DISCUSSION TITLE
OS.js with pin-website in taskbar?

### DISCUSSION THREAD

**Discussion by ghost** [os-js/OS.js#818] [Q&A] [answered]
Hi all.

#### 1. concept
I would like to be able to add a website to the taskbar, something similar to windows 10 in os.js. Please read this tutorial in [how-to-pin-a-website-to-the-windows-10-taskbar-or-start-menu](https://www.howtogeek.com/656737/how-to-pin-a-website-to-the-windows-10-taskbar-or-start-menu/) for more information.

#### 2.1 idea 1.1
![image](https://user-images.githubusercontent.com/111701513/206925732-e36f531d-cc77-48d7-8d39-fc8032dd67ec.png)

#### 2.2 idea 1.2
![image](https://user-images.githubusercontent.com/111701513/206925762-cdbfe826-c30b-49f4-8e5d-f424f811d549.png)

#### 2.3 idea 1.3
![image](https://user-images.githubusercontent.com/111701513/206925804-b1286a14-15d3-4dbf-a15b-f7e6c06e0891.png)

#### 3.1 ui/ux, *"os.js with pin taskbar?"*
![image](https://user-images.githubusercontent.com/111701513/206926418-ee1f0bf1-fa90-4f5c-9806-cd0bcf15d985.png)

#### 3.2 "source-code", *"os.js with pin taskbar?"*
```html
<div class="osjs-system-context-menu">
   <div id="osjs-context-menu" class="osjs-gui osjs-gui-menu" style="display: block; top: 146px; left: 180px;">
      <ul class="">
         <li class="osjs-gui-menu-entry">
            <div class="osjs-gui-menu-container"><span class="osjs-gui-menu-label "><span>Select wallpaper</span></span></div>
         </li>
         <li class="osjs-gui-menu-entry">
            <div class="osjs-gui-menu-container" data-has-children="true">
               <span class="osjs-gui-menu-label "><span>Select theme</span></span>
               <ul class="">
                  <li class="osjs-gui-menu-entry">
                     <div class="osjs-gui-menu-container"><span class="osjs-gui-menu-label "><span>Standard Dark Theme</span></span></div>
                  </li>
                  <li class="osjs-gui-menu-entry">
                     <div class="osjs-gui-menu-container"><span class="osjs-gui-menu-label "><span>Standard Theme</span></span></div>
                  </li>
                  <li class="osjs-gui-menu-entry">
                     <div class="osjs-gui-menu-container"><span class="osjs-gui-menu-label "><span> Windows 8 Style Theme</span></span></div>
                  </li>
               </ul>
            </div>
         </li>
         <li class="osjs-gui-menu-entry">
            <div class="osjs-gui-menu-container" data-has-children="true">
               <span class="osjs-gui-menu-label "><span>Add Widget</span></span>
               <ul class="">
                  <li class="osjs-gui-menu-entry">
                     <div class="osjs-gui-menu-container"><span class="osjs-gui-menu-label "><span>Digital Clock</span></span></div>
                  </li>
               </ul>
            </div>
         </li>
         <li class="osjs-gui-menu-entry">
            <div class="osjs-gui-menu-container" data-has-children="true">
               <span class="osjs-gui-menu-label "><span>Add shortcut in taskbar</span></span>
               <ul class="">
                  <li class="osjs-gui-menu-entry">
                     <div class="osjs-gui-menu-container"><span class="osjs-gui-menu-label "><span>Shortcut taskbar</span></span></div>
                  </li>
               </ul>
            </div>
         </li>
      </ul>
   </div>
</div>
```
#### 4. Question/feedback
*Have you all thought about this idea?*

**Comment by andersevenrud** [comment, marked as answer]
Sounds like you might want to look into creating a panel item where you can add custom launchers.

https://manual.os-js.org/tutorial/panel/

This is possible to integrate with drag-and-drop and context menus in the file manager.

**Reply by ghost** [reply]
Hi andersevenrud.

> Sounds like you might want to look into creating a panel item where you can add custom launchers.

Yes, exactly.

> This is possible to integrate with drag-and-drop and context menus in the file manager.

thank you for feedback.
